scripts/Player.gd

```diff
@@ -134,3 +155,107 @@ func selectPawnUsingFallbackStrategy():
 		while (not pawns[choosen].isPawnInHome()):
 			choosen = randi() % 4
 		choosenPawn = choosen
+
+func selectPawnUsingSoloStrategy():
+	printDebug()
+	if (isAnyPawnOnBoard()):
+		selectTheFurthestPawn()
+	else:
+		if board.diceResult == 6:
+			var isSelected = selectPawnToLeaveHome()
+			if (not isSelected):
+				fallbackToFallbackStrategy("solo (3)")
+		else:
+			fallbackToFallbackStrategy("solo (1)")
+
+func selectTheFurthestPawn():
+	var status = getPawnStatus()
+	var furthestDistance = -11
+	var furthestPawn = -1
+	for pawnId in range(4):
+		if pawns[pawnId].isPawnOnBoard():
+			if furthestDistance < status[pawnId].y:
+				furthestDistance = status[pawnId].y
+				furthestPawn = pawnId
+	if furthestPawn != -1:
+		choosenPawn = furthestPawn
+	else:
+		fallbackToFallbackStrategy("solo (2)")
+
+func getPawnStatus():
+	var status = []
+	for pawnId in range(4):
+		status.append(Vector2(pawns[pawnId].pawnPlace, pawns[pawnId].distanceFromStart))
+	return status
+
+func fallbackToFallbackStrategy(strategyName):
+	print("ERROR: "  + str(strategyName) + " strategy had a problem and stopped working, using fallback strategy")
+	printDebug(true)
+	selectPawnUsingFallbackStrategy()
+
+func selectPawnUsingBalancedStrategy():
+	printDebug()
+	if board.diceResult == 6:
+		selectTheNearestPawn()
+	else:
+		selectTheNearestPawnFromBoard()
+
+func selectTheNearestPawn():
+	var status = getPawnStatus()
+	var nearestDistance = 40
+	var nearestPawn = -1
```

Comment:
Is '-1' safe? What if two pawns leave home and the second of them be placed on -2 field.

Probably better to put something like 'undefined' string value. I hope there won't be any value conflicts.